.github/workflows/ci.yml

```diff
@@ -29,45 +29,135 @@ jobs:
             cppyy: Off
             extra_cmake_options: '-DLLVM_USE_LINKER=lld'
 
+          - name: ubu22-gcc12-clang-repl-17-cppyy
+            os: ubuntu-latest
+            compiler: gcc-12
+            clang-runtime: '17'
+            cling: Off
+            cppyy: On
+            extra_cmake_options: '-DLLVM_USE_LINKER=lld'
+
           - name: ubu22-gcc9-clang-repl-16
-            os: ubuntu-22.04
+            os: ubuntu-latest
             compiler: gcc-9
             clang-runtime: '16'
             cling: Off
             cppyy: Off
             coverage: true
             extra_cmake_options: '-DLLVM_USE_LINKER=lld'
 
+          - name: ubu22-gcc9-clang-repl-16-cppyy
+            os: ubuntu-latest
+            compiler: gcc-9
+            clang-runtime: '16'
+            cling: Off
+            cppyy: On
+            coverage: true
+            extra_cmake_options: '-DLLVM_USE_LINKER=lld'
+
           - name: ubu22-gcc9-clang13-cling
-            os: ubuntu-22.04
+            os: ubuntu-latest
             compiler: gcc-9
             clang-runtime: '13'
             cling: On
             cppyy: Off
             extra_cmake_options: '-DLLVM_USE_LINKER=lld'
 
           - name: ubu22-gcc9-clang13-cling-cppyy
-            os: ubuntu-22.04
+            os: ubuntu-latest
             compiler: gcc-9
             clang-runtime: '13'
             cling: On
             cppyy: On
             extra_cmake_options: '-DLLVM_USE_LINKER=lld'
 
-          - name: osx-clang-clang-repl-16
-            os: macos-latest
+          - name: osx-x86-clang-clang-repl-17
+            os: macos-13
+            compiler: clang
+            clang-runtime: '17'
+            cling: Off
+            cppyy: Off
+
+          - name: osx-x86-clang-clang-repl-17-cppyy
+            os: macos-13
+            compiler: clang
+            clang-runtime: '17'
+            cling: Off
+            cppyy: On
+
+          - name: osx-x86-clang-clang-repl-16
+            os: macos-13
             compiler: clang
             clang-runtime: '16'
             cling: Off
             cppyy: Off
 
-          - name: osx-clang-clang13-cling
-            os: macos-latest
+          - name: osx-x86-clang-clang-repl-16-cppyy
+            os: macos-13
+            compiler: clang
+            clang-runtime: '16'
+            cling: Off
+            cppyy: On
+
+          - name: osx-x86-clang-clang13-cling
+            os: macos-13
             compiler: clang
             clang-runtime: '13'
             cling: On
             cppyy: Off
 
+          - name: osx-x86-clang-clang13-cling-cppyy
+            os: macos-13
+            compiler: clang
+            clang-runtime: '13'
+            cling: On
+            cppyy: On
+
+          #Block commented out until free tier for m1
+          #exists (expected sometime 2024) and key for os 
+          #can be replaced
+          #- name: osx-arm-clang-clang-repl-17
```

Comment:
Maybe in the name it is better to put arm64 instead of just arm.

Reply:
I've now made this change, as well as change osx to osx13 to make it clear its running on macos version 13.